kobo/apps/audit_log/models.py

```diff
@@ -335,13 +335,55 @@ def create_from_request(cls, request):
             'asset-file-list': cls.create_from_file_request,
             'asset-export-list': cls.create_from_export_request,
             'exporttask-list': cls.create_from_v1_export,
+            'asset-bulk': cls.create_from_bulk_request,
         }
         url_name = request.resolver_match.url_name
         method = url_name_to_action.get(url_name, None)
         if not method:
             return
         method(request)
 
+    @staticmethod
+    def create_from_bulk_request(request):
+        try:
+            payload = request._data['payload']
+            action = payload['action']
+            asset_uids = payload['asset_uids']
+        except KeyError:
+            return  # Incorrect payload
+
+        if type(asset_uids) is not list or len(asset_uids) == 0:  # Nothing to do
+            return
+
+        bulk_action_to_audit_action = {
+            'archive': AuditAction.ARCHIVE,
+            'unarchive': AuditAction.UNARCHIVE,
+            'delete': AuditAction.DELETE,
+            'undelete': AuditAction.UNDELETE,
+        }
+        audit_action = bulk_action_to_audit_action[action]
+        if audit_action is None:
+            return  # Unsupported action
+
+        source = get_human_readable_client_user_agent(request)
+        client_ip = get_client_ip(request)
+
+        for asset_uid in asset_uids:
+            asset = Asset.all_objects.get(uid=asset_uid)
+            object_id = asset.pk
+            metadata = {
+                'asset_uid': asset_uid,
```

Comment:
we should also store the latest_version_uid for the archive/unarchive requests. See the `deployment` endpoint for how we do this (basically add the relevant information to request._request).